versioned/storage/common-tests/src/main/java/org/projectnessie/versioned/storage/commontests/AbstractIndexesLogicTests.java

```diff
@@ -338,4 +341,68 @@ public void perCommitOps() {
 
     soft.assertThat(indexesLogic.commitOperations(commit)).containsExactly(add1, add2, remove1);
   }
+
+  @Test
+  void completeIndexesInCommitChainWithStripes() throws Exception {
+
+    // Create 10 commits with 100 PUTs each;
+    // Commit N is thus expected to have 100 * N index entries in total.
+    CommitObj parent = null;
+    for (int i = 1; i <= 10; i++) {
+
+      ObjId parentCommitId = parent == null ? EMPTY_OBJ_ID : parent.id();
+
+      CommitObj.Builder c =
+          commitBuilder()
+              .id(randomObjId())
+              .addTail(parentCommitId)
+              .created(System.nanoTime())
+              .seq(i)
+              .message("Commit " + i)
+              .headers(EMPTY_COMMIT_HEADERS)
+              .incompleteIndex(true);
+      StoreIndex<CommitOp> index = newStoreIndex(CommitOp.COMMIT_OP_SERIALIZER);
+
+      Random random = new Random();
+
+      for (int j = 0; j < 100; j++) {
+
+        // Use large keys to increase chances of ObjTooLargeException which
+        // in turn will trigger the striped index logic.
+        String key =
+            random
+                .ints('0', 'z' + 1)
+                .limit(100)
```

Comment:
The `Random` approach can, although unlikely, produce duplicate keys.
Why not go with something like `String key = "x"+i+SOME_LONG_CONSTANT`, which effectively eliminates prefix-encoding in the serialized index?